### PR TITLE
Updates to the beamforming core, from the top and down

### PR DESCRIPTION
**Pull request by mofii**
This refactores how the code is running, making it clearer to read and also closer to actually being able to run ;)

There are still some obvious issues (such as working in-place on a vector element which won't be reflected in the original vector), I'll handle them in later PRs.

### CODE REVIEW

**Comment by mofii** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1** <a href="https://app.graphite.dev/github/pr/Orchard-Ultrasound-Innovation/Beamformer.jl/1" target="_blank">Graphite</a> 
    * **PR #2** <a href="https://app.graphite.dev/github/pr/Orchard-Ultrasound-Innovation/Beamformer.jl/2" target="_blank">Graphite</a> 
      * **PR #4** <a href="https://app.graphite.dev/github/pr/Orchard-Ultrasound-Innovation/Beamformer.jl/4" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/Orchard-Ultrasound-Innovation/Beamformer.jl/4?utm_source=stack-comment).

File: src/apodizations.jl [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
+function rcv_apodization!(px_apo, px, rcv_elms)
+    for (i,rcv_elm) in rcv_elms
+        px_apo[i] = px_rcv_apo(px, rcv_elm)
+    end
+    return nothing
+end
```
Comment (stevenaleung) [inline src/apodizations.jl:7]: What is `px` short for?

Comment (stevenaleung) [reply src/apodizations.jl:7]: I see, it's pixels. Thought it was an acronym or shortened like `transmit` -> `tx`. Didn't understand until I saw `Bmode.pxs`. Could we use `pix` or `pixel` instead for a little more clarity?

File: src/das_beamformer.jl [hunk 2]
```diff
@@ -1,63 +1,92 @@
...
 
 # TODO:
 # - setup interpolators before calling this function.
-# - make struct that hold information about the RF data (t_start etc)
-function delay_sum_and_apodize!(px_vals, px_idx, rf_data, data_t_start, data_dt, delays, rcv_apo, src_apo)
-    num_channels = size(rf_data,2)
-    vals = Vector{Complex{Float64}}(undef, num_channels)
+"""
+$(SIGNATURES)
+"""
```
Comment (stevenaleung) [inline src/das_beamformer.jl:64]: This is causing the CI to fail. Probably just need to add `using DocStringExtentions` but that line is included upstack in #5, so not sure if you want to remove it from #5 after adding it here. Even if there end up being two lines for `using DocStringExtensions` it probably won't be a problem

**Review by stevenaleung — COMMENTED** [review]
As we discussed earlier, I skimmed the following and will review at a later time when they're working:
- `rcv_apodization!()`
- `das_beamform_bmode!()`
- `das_beamform_tx_event!()`
- `das_beamform_px!()`
- `calculate_delays!()`
- `get_source_to_focus_to_pixel()`

File: src/apodizations.jl (hunk 1 above)
Comment (mofii) [reply src/apodizations.jl:7]: Done

**Comment by mofii** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Stack job ID: XDQOyRXr3zhF1tW9o6yg.
See details on [graphite.dev](https://app.graphite.dev/github/pr/Orchard-Ultrasound-Innovation/Beamformer.jl/4)

**Pull request merged**